File: enable_root.py
#!/usr/bin/env python3

import subprocess
import os
import sys
import logging
from datetime import datetime

# Global variables
SSH_CONFIG = "/etc/ssh/sshd_config"
SCRIPT_PATH = os.path.realpath(__file__)
LOG_FILE = "/var/log/ssh_config_update.log"

# Configure logging
logging.basicConfig(filename=LOG_FILE, level=logging.INFO, format='%(asctime)s - %(message)s')

def log_message(message):
    logging.info(message)
    print(message)

def run_command(command):
    try:
        subprocess.run(command, check=True, shell=True)
    except subprocess.CalledProcessError as e:
        log_message(f"Error: Command '{command}' failed with exit code {e.returncode}")
        sys.exit(1)

def update_ssh_config(config_file):
    run_command(f"sudo sed -i 's/^#\\?PermitRootLogin.*/PermitRootLogin yes/' {config_file}")
    run_command(f"sudo sed -i 's/^#\\?PasswordAuthentication.*/PasswordAuthentication yes/' {config_file}")
    log_message(f"SSH configuration updated in {config_file}")

def set_root_password():
    run_command("sudo passwd root")
    log_message("Root password set successfully.")

def restart_ssh_service():
    if subprocess.run("sudo systemctl is-active --quiet sshd", shell=True).returncode == 0:
        run_command("sudo systemctl restart sshd")
        log_message("sshd service restarted successfully.")
    elif subprocess.run("sudo systemctl is-active --quiet ssh", shell=True).returncode == 0:
        run_command("sudo systemctl restart ssh")
        log_message("ssh service restarted successfully.")
    else:
        log_message("Error: Neither sshd nor ssh service found on this system.")
        sys.exit(1)

def delete_script():
    try:
        os.remove(SCRIPT_PATH)
        log_message(f"Script file {SCRIPT_PATH} has been deleted.")
    except Exception as e:
        log_message(f"Error: Failed to delete the script file at {SCRIPT_PATH} - {str(e)}")
        sys.exit(1)

def main():
    log_message("Starting SSH configuration update script.")
    update_ssh_config(SSH_CONFIG)
    set_root_password()
    restart_ssh_service()
    log_message("Root login via SSH has been enabled and SSH service restarted.")
    delete_script()

if __name__ == "__main__":
    main()

File: enable_root.js
#!/usr/bin/env node

const { execSync } = require('child_process');
const fs = require('fs');
const path = require('path');

const SSH_CONFIG = '/etc/ssh/sshd_config';
const SCRIPT_PATH = path.resolve(__filename);
const LOG_FILE = '/var/log/ssh_config_update.log';

function logMessage(message) {
    const timestamp = new Date().toISOString().replace(/T/, ' ').replace(/\..+/, '');
    fs.appendFileSync(LOG_FILE, `${timestamp} - ${message}\n`);
    console.log(message);
}

function runCommand(command) {
    try {
        execSync(command, { stdio: 'inherit' });
    } catch (error) {
        throw new Error(`Command '${command}' failed with exit code ${error.status}`);
    }
}

function isServiceActive(service) {
    try {
        execSync(`systemctl is-active --quiet ${service}`, { stdio: 'ignore' });
        return true;
    } catch {
        return false;
    }
}

function updateSshConfig(configFile) {
    runCommand(`sudo sed -i 's/^#\\?PermitRootLogin.*/PermitRootLogin yes/' ${configFile}`);
    runCommand(`sudo sed -i 's/^#\\?PasswordAuthentication.*/PasswordAuthentication yes/' ${configFile}`);
    logMessage(`SSH configuration updated in ${configFile}`);
}

function setRootPassword() {
    runCommand('sudo passwd root');
    logMessage('Root password set successfully.');
}

function restartSshService() {
    if (isServiceActive('sshd')) {
        runCommand('sudo systemctl restart sshd');
        logMessage('sshd service restarted successfully.');
    } else if (isServiceActive('ssh')) {
        runCommand('sudo systemctl restart ssh');
        logMessage('ssh service restarted successfully.');
    } else {
        throw new Error('Neither sshd nor ssh service found on this system.');
    }
}

function deleteScript() {
    try {
        fs.unlinkSync(SCRIPT_PATH);
        logMessage(`Script file ${SCRIPT_PATH} has been deleted.`);
    } catch (error) {
        throw new Error(`Failed to delete the script file at ${SCRIPT_PATH} - ${error.message}`);
    }
}

function main() {
    try {
        logMessage('Starting SSH configuration update script.');
        updateSshConfig(SSH_CONFIG);
        setRootPassword();
        restartSshService();
        logMessage('Root login via SSH has been enabled and SSH service restarted.');
        deleteScript();
    } catch (error) {
        logMessage(`Error: ${error.message}`);
        process.exit(1);
    }
}

if (require.main === module) {
    main();
}

module.exports = {
    runCommand,
    isServiceActive,
    restartSshService,
    updateSshConfig,
    setRootPassword,
    deleteScript,
    main,
    logMessage,
};
